### PR TITLE
Use uid instead of id or uuid

### PR DESCRIPTION
**Pull request by itziakos**
This PR updates the abstract description and native implementation of the Particles and Mesh containers to use `uid` instead of `id` or `uuid`

### CODE REVIEW

**Comment by itziakos** [comment]
This fixes #20, @roigcarlo  or @nathanfranklin please have a look

File: simphony/io/file_particle_container.py [hunk 1]
```diff
@@ -253,14 +253,14 @@ def _bond_to_row(self, bond, id):
...
         particle_ids[:n] = bond.particles
-        return id, particle_ids, n
+        return uid, particle_ids, n
 
     def _generate_unique_id(self, table, number_tries=1000):
         for n in xrange(number_tries):
-            id = random.randint(0, MAX_INT)
-            for _ in table.where('id == value', condvars={'value': id}):
+            uid = random.randint(0, MAX_INT)
+            for _ in table.where('id == value', condvars={'value': uid}):
```
Comment (nathanfranklin) [inline simphony/io/file_particle_container.py]: ´uid == value`

Comment (itziakos) [inline simphony/io/file_particle_container.py]: Perfect, it looks that the tests do not check that :smile:

Comment (itziakos) [inline simphony/io/file_particle_container.py]: I will fix the tests in #33

Comment (itziakos) [inline simphony/io/file_particle_container.py]: typo is fixed

**Comment by nathanfranklin** [comment]
:+1:

**Pull request merged**